### PR TITLE
Handlers Utils 

### PR DESCRIPTION
**Pull request by m26unkwn**
1. Extract all the function from he components

### CODE REVIEW

**Review by PuneetP16 — APPROVED** [review]
Appreciation: Good work on making code more clean. Looks good to merge. Looks more easier to manage individual components by doing this.

File: src/components/PrivateRoute/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const PrivateRoute = ({ children }) => {
   return token ? (
     children
   ) : (
-    <Navigate to='/login' state={{ from: location }} replace />
```
Comment (tarunsankhla) [inline src/components/PrivateRoute/index.js:14]: Create const of routes for better maintainability

File: src/components/authentication/Login/index.js [hunk 2]
```diff
@@ -23,18 +24,18 @@ const Login = () => {
...
 
-  const onFocusHandler = (type) => {
+  const onFocus = (type) => {
     dispatch({ type, payload: false });
   };
 
-  const testLoginHandler = (e) => {
+  const testUserLogin = (e) => {
     e.preventDefault();
     getUserAuth("/api/auth/login", "adarshbalika@gmail.com", "Adarshbalika1!");
```
Comment (tarunsankhla) [inline src/components/authentication/Login/index.js:40]: Suggestion Create const of path

File: src/components/cart/index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { useData } from "../../Context/stateManage/state-context";
+import { useData } from "../../Context";
```
Comment (tarunsankhla) [inline src/components/cart/index.js:1]: Suggestion try using absolute path

File: src/utils/handlers.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  dispatch,
+  setLoading,
+  isProducInWishlist
+) => {
+  const header = { authorization: token };
+  if (isProducInWishlist) {
+    alert("Item already inside of Wishlist");
+  } else {
+    getDataFromServer(
+      `/api/user/wishlist`,
```
Comment (tarunsankhla) [inline src/utils/handlers.js:34]: Same here const

**Review by tarunsankhla — APPROVED** [review]
Overall great just minor changes

File: src/utils/handlers.js (hunk 4 above)
Comment (m26unkwn) [reply src/utils/handlers.js:34]: Ok will try do all the require changes

**Review by saurabhchirde — APPROVED** [review]
Liked the way you have refactored your code, it is pretty clean and easy to read now. 
Good to merge.

**Pull request merged**